### PR TITLE
SNOW-1374305: Add docstring to Dataframe.merge method

### PR DESCRIPTION
**Pull request by sfc-gh-nkumar**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1374305

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Added docstring to Dataframe.merge method

### CODE REVIEW

File: src/snowflake/snowpark/modin/pandas/dataframe.py
Comment (sfc-gh-joshi) [inline src/snowflake/snowpark/modin/pandas/dataframe.py:1]: You'll need to put this docstring in modin/plugin/docstrings/dataframe.py to get it to show up. Can you also double check the built docs (`cd docs; make html; open build/html/index.html`) to make sure everything looks fine?

https://github.com/snowflakedb/snowpark-python/blob/febe20b41bfe208abc90e1de20a9ec9b60072f89/src/snowflake/snowpark/modin/plugin/docstrings/dataframe.py#L1890-L1891

File: src/snowflake/snowpark/modin/pandas/dataframe.py [hunk 1]
```diff
@@ -1627,6 +1627,176 @@ def merge(
...
+            If True, adds a column to the output DataFrame called "_merge" with
+            information on the source of each row. The column can be given a different
+            name by providing a string argument. The column will have a Categorical
+            type with the value of "left_only" for observations whose merge key only
+            appears in the left DataFrame, "right_only" for observations
+            whose merge key only appears in the right DataFrame, and "both"
+            if the observation's merge key is found in both DataFrames.
+
+        validate : str, optional
+            This is not yet supported.
```
Comment (sfc-gh-joshi) [inline src/snowflake/snowpark/modin/pandas/dataframe.py:1704]:
```suggestion
            This argument is not yet supported in Snowpark pandas.
```

**Review by sfc-gh-joshi — CHANGES_REQUESTED** [review]

**Review by sfc-gh-mvashishtha — APPROVED** [review]
LGTM apart from @sfc-gh-joshi 's comments

File: src/snowflake/snowpark/modin/pandas/dataframe.py
Comment (sfc-gh-nkumar) [reply src/snowflake/snowpark/modin/pandas/dataframe.py:1]: Ah forgot that we have plugin for docs now. Fixed it.

**Review by sfc-gh-joshi — APPROVED** [review]
I see a couple of warnings when building docs https://github.com/snowflakedb/snowpark-python/actions/runs/9069683454/job/24919875813?pr=1566
```
/home/runner/work/snowpark-python/snowpark-python/src/snowflake/snowpark/modin/pandas/dataframe.py:docstring of snowflake.snowpark.modin.pandas.dataframe.DataFrame.merge:19: ERROR: Unexpected indentation.
/home/runner/work/snowpark-python/snowpark-python/src/snowflake/snowpark/modin/pandas/dataframe.py:docstring of snowflake.snowpark.modin.pandas.dataframe.DataFrame.merge:20: WARNING: Block quote ends without a blank line; unexpected unindent.
```

LGTM otherwise.

**Review by sfc-gh-oplaton — APPROVED** [review]

**Pull request merged**